aten/src/ATen/core/dispatch/OperatorMetadata.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+
+namespace c10 {
+namespace detail {
+
+template<class Data> class OperatorMetadataType;
+
+template<class Data>
+class OpRegistrationListenerForMetadata final : public OpRegistrationListener {
+public:
+  explicit OpRegistrationListenerForMetadata(OperatorMetadataType<Data>* metadataType)
+  : metadataType_(metadataType) {}
+
+  void onOperatorRegistered(const OperatorHandle& op) override {
+    // no-op
+  }
+
+  void onOperatorDeregistered(const OperatorHandle& op) override {
+    metadataType_->removeOperator(op);
+  }
+
+private:
+  OperatorMetadataType<Data>* metadataType_;
+};
+
+/**
+ * OperatorMetadataType is an extensible way to store metadata for an operator
```

Comment:
I think this is over-engineered. We _don't want_ people adding arbitrary stuff here, we want to just have storage for the specific things we know we care about.

Meta-data is a code smell. There is either data we understand and can name. Or there is data we do not know anything about, and should not be in the business of storing.